### ISSUE TITLE
Filter range in main page

### ISSUE THREAD

**Issue by atomrab** [periodo/periodo-client#57]
What's determining the year range of the filter at the upper right? I love this as a strategy, but using the web backend, it only shows -50000 to 2100 for the dataset as a whole. If this is just for the first page, it doesn't adjust with subsequent pages. If this is for the whole set, the earliest date is too late (we go back to 1.5mya already, and soon will go back a lot further with geological stratigraphy).

**Comment by ptgolden** [comment]
You can brush the range to select a window.

>  If this is just for the first page, it doesn't adjust with subsequent pages.

I believe it does.

> If this is for the whole set, the earliest date is too late (we go back to 1.5mya already, and soon will go back a lot further with geological stratigraphy). 

This was on purpose. I originally had it so it was the entire range, but then 99% of the periods were in the topmost range. This seemed a lot more useful. If you have another suggestion I'd be happy to hear it.

**Comment by atomrab** [comment]
I'll check the subsequent pages -- when I last tried it, it didn't adjust
for the first two, but maybe I didn't go far enough.

As for the range, would it be possible to include a discrete zoom button to
change the scale, so you could zoom out to a geological level or in for a
historical one? I understand the dilemma, and it's not major right now, but
the geological eras are going to be tougher to find if this filter (which
is otherwise extremely effective) wont work for them.

**Comment by ptgolden** [comment]
> As for the range, would it be possible to include a discrete zoom button to change the scale, so you could zoom out to a geological level or in for a historical one?

Yes, that would be a very useful feature. One might be able to switch between a logarithmic and linear time scale. Or, the degree to which outliers are dropped could be configured/disabled.

Currently, most of our periods are at human, not geological, scale. I tried to focus on making a filter for those.

**Comment by atomrab** [comment]
On Wednesday, July 8, 2015, Patrick Golden <notifications@github.com> wrote:

> As for the range, would it be possible to include a discrete zoom button
> to change the scale, so you could zoom out to a geological level or in for
> a historical one?
>
> Yes, that would be a very useful feature. One might be able to switch
> between a logarithmic and linear time scale. Or, the degree to which
> outliers are dropped could be configured/disabled.
>
So do you think that this is implementable now, or next phase?


> Currently, most of our periods are at human, not geological, scale. I
> tried to focus on making a filter for those.
>

Yup, makes sense. The geological ones are queued to go in as soon as this
version of the client is ready to go, which is why it was on my mind. And
there's a UT project related to the fossil record for early humans that
wants to use PeriodO for its periods, so this will come up sooner rather
than later.

**Comment by ptgolden** [comment]
> So do you think that this is implementable now, or next phase?

To make something that would be appropriate in all situations would take a good amount of time. There will always be improvements that could be made. I'll see if there are any easy ways around this tomorrow, but in general- yes, next phase.

**Comment by rybesh** [comment]
What about leaving it as is, but adding something like "3 outliers not displayed" under where it says "Viewing from -49942 to 2100", and clicking on that would zoom out to show everything.

Also, if I understand correctly, the zoom is set based on how skewed the selected set is, so if someone is looking at a set that consists of all geological-scale definitions, it wouldn't drop any, right?

**Comment by atomrab** [comment]
I like Ryan's solution, especially since it seems like it can be implemented in the short term.

The zoom set definitely scales to a set selected in the filter (e.g. select Fasti, and the scale drops back to 1mya). But it does not change as you move from page to page in the unfiltered set -- the range stays exactly the same, regardless of the periods being displayed. 

I also just found some strange behavior in Chrome on Windows 7: if you select Fasti, change the temporal filter range from around a 1mya to around 500kya (producing a result set with mainly lower Paleolithic), and then hit "reset" in the source filter, you only see the 23 periods from all sources that overlap with the timespan filter for the Fasti dataset. But since the temporal filter at that point only runs from 50kya forward, you can't modify the selection without resetting the filter to 50kya forward (though this seems to continue to include those paleolithic entries, since our total is 1763 right now, right?).

**Comment by atomrab** [comment]
I would also like to repeat my preference for displaying dates in the UI front page in standardized ISO form, not with the original label. I understand the reasoning for displaying the original label, but right now, without opening a specific period for details, it's impossible to distinguish between ISO8601 positive 1776 = A.D. 1776 (as in Open Context/DINAA) and 1776 (BP 2000) = A.D. 224. This is going to be a real user problem, especially since we'll have a ton of dates from ARIADNE that are all in the idiosyncratic Fasti form (which we don't have a marker for -- we say "B.C." and "BCE", but we dropped the "BP (2000)" tag that attached to the Fasti values).

**Comment by atomrab** [comment]
Actually, the total number of entries on load is 1761, but when you do the series of actions I describe with Fasti above, you end up with a total of 1763. What does that mean?

**Comment by ptgolden** [comment]
> Also, if I understand correctly, the zoom is set based on how skewed the selected set is, so if someone is looking at a set that consists of all geological-scale definitions, it wouldn't drop any, right?

Right

**Comment by ptgolden** [comment]
> But it does not change as you move from page to page in the unfiltered set -- the range stays exactly the same, regardless of the periods being displayed.

This is not a bug. The pagination is showing all periods matched by the current filter criteria. It's not the other way around- as in, the filter criteria reflecting the 20 periods currently displayed.

**Comment by ptgolden** [comment]
>  But since the temporal filter at that point only runs from 50kya forward, you can't modify the selection without resetting the filter to 50kya forward (though this seems to continue to include those paleolithic entries, since our total is 1763 right now, right?). 

I see what's going on and I'll fix it when I make the outlier handling more explicit.

**Comment by atomrab** [comment]
So does this mean that no periods that end before 50kya are displayed when
you load the client? How will people know they're there, if they don't
already have a source that they plan to consult?

**Comment by atomrab** [comment]
Also, what is our total number of values right now? 1791 appear on load,
without additional filtering, but from what you just said, this does not
include the pre-50kya end date periods, since those aren't in the temporal
filter (or maybe the temporal filter just ignores them when you load all?).
1763 appear after the weird sequence I described with Fasti, but these
definitely include the pre-50kya end date periods.

**Comment by ptgolden** [comment]
> I would also like to repeat my preference for displaying dates in the UI front page in standardized ISO form, not with the original label.

ok, done

**Comment by ptgolden** [comment]
> So does this mean that no periods that end before 50kya are displayed when you load the client? 

no. I'm aware of the issue you raised and I will fix it.

**Comment by ptgolden** [comment]
Alright, I took out the confusing "viewing from ... " message which just repeated the information at the ends of the scale.

I also added a "hide outliers" option that will turn off "zooming in" to a section where the vast majority of periods are located.

I also fixed the bug where the brush would still be in effect even after not being visible.

I don't have time to do anything more complicated than changing the label for the checkbox. The text involved is here: https://github.com/periodo/periodo-client/blob/master/src/components/shared/range_selection.jsx#L113-L126 if anyone wants to change it.
